### PR TITLE
GEODE-10417: Fix NullPointerException in WAN replication

### PR DESCRIPTION
**Pull request by albertogpz**
When the WAN group-transaction-events feature is enabled in
a parallel gateway sender, it is possible to get a NullPointerException
when retrieving events from the queue to complete a transaction if
the event in the queue is null.

If this situation is reached then the gateway sender dispatcher will
not dispatch queue events anymore and therefore the WAN replication
will not progress.

This happens because the predicates that check if elements
in the queue contain a transactionId are not protected
against the event being null.

A null check has been added before the predicates are invoked
so that in case of a null event, the predicate is not invoked
and the event is skipped from the checking.

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

**Review by mivanac — COMMENTED** [review]
I think that this is needed check, since after all other calls of optimalGet(), we check if object is null.

**Review by mivanac — APPROVED** [review]

**Comment by jvarenina** [comment]
No comments from my side. Good work Alberto!

**Review by mivanac — CHANGES_REQUESTED** [review]
I just remember, that we should use assertThat instead of assertEquals

File: geode-core/src/test/java/org/apache/geode/internal/cache/BucketRegionQueueJUnitTest.java [hunk 1]
```diff
@@ -191,6 +191,49 @@ public void testGetElementsMatchingWithParallelGatewaySenderQueuePredicatesAndSo
...
+    Predicate<GatewaySenderEventImpl> hasTransactionIdPredicate =
+        ParallelGatewaySenderQueue.getHasTransactionIdPredicate(tx1);
+    Predicate<GatewaySenderEventImpl> isLastEventInTransactionPredicate =
+        ParallelGatewaySenderQueue.getIsLastEventInTransactionPredicate();
+    when(bucketRegionQueue.getValueInVMOrDiskWithoutFaultIn(4L)).thenReturn(null);
+    List<Object> objects = this.bucketRegionQueue.getElementsMatching(hasTransactionIdPredicate,
+        isLastEventInTransactionPredicate);
+
+    assertEquals(2, objects.size());
+    assertEquals(objects, Arrays.asList(new Object[] {event1, event7}));
```
Comment (mivanac) [inline geode-core/src/test/java/org/apache/geode/internal/cache/BucketRegionQueueJUnitTest.java:233]: Please replace all with AssertJ.

**Review by mivanac — APPROVED** [review]

**Pull request merged**